forest/daemon/src/daemon.rs

```diff
@@ -392,6 +395,27 @@ pub(super) async fn start(config: Config, detached: bool) -> Db {
     db
 }
 
+// returns the first error with which any of the services end
+// in case all services finished without an error sleeps for more than 2 years
+// and then returns with an error
+async fn propagate_error(services: &mut JoinSet<Result<(), anyhow::Error>>) -> anyhow::Error {
+    while !services.is_empty() {
+        select! {
+            option = services.join_next().fuse() => {
+                if let Some(Ok(Err(error_message))) = option {
+                    return error_message
+                }
+            },
+        }
+    }
+    // In case all services are down without errors we are still willing
+    // to wait indefinitely for CTRL-C signal. As `tokio::time::sleep` has
+    // a limit of approximately 2.2 years we have to loop
+    loop {
+        tokio::time::sleep(Duration::new(64000000, 0)).await;
+    }
```

Comment:
I think after 2 years of waiting we are fine to retire this process...

Reply:
David explains in the upper comments that a situation in which we should wait indefinitely